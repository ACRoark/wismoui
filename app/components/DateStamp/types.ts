export type DateStampFormat = 'auto' | 'long' | 'short';
