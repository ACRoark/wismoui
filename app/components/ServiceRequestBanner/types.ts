export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
