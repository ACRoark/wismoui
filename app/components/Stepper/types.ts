import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
