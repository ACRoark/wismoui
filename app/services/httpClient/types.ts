export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
