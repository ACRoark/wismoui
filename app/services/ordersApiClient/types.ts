import { IHttpError } from 'services/httpClient';
import { IOrder } from 'types';
import IOrderSearchResults from 'types/IOrderSearchResults';
import IOrderSearchResultsViewModel from 'types/IOrderSearchResultsViewModel';
import IOrderViewModel from 'types/IOrderViewModel';

export interface IFindOrderResponse {
  error?: IHttpError;
  order: IOrder;
  viewModel?: IOrderViewModel;
}

export interface IGetOrderResponse {
  error?: IHttpError;
  order: IOrder;
  viewModel?: IOrderViewModel;
}

export interface ISearchOrdersResponse {
  error?: IHttpError;
  results: IOrderSearchResults;
  viewModel?: IOrderSearchResultsViewModel;
}
