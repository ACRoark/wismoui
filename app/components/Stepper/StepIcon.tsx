import React, { FC, ReactElement } from 'react';

import CompletedIcon from 'components/Icons/CompletedIcon';
import ErrorIcon from 'components/Icons/ErrorIcon';

import { StepperStatus } from './types';

interface IStepIcon {
  status: StepperStatus;
}

const StepIcon: FC<IStepIcon> = ({ status }: IStepIcon): ReactElement => {
  switch (status) {
    case StepperStatus.Active:
      return <i className="icon active" />;
    case StepperStatus.Completed:
      return <CompletedIcon />;
    case StepperStatus.Error:
      return <ErrorIcon />;
    case StepperStatus.Pending:
    default:
      return <i className="icon pending" />;
  }
};

export default StepIcon;
