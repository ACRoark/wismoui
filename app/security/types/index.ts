import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
