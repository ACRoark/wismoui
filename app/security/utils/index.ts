import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
