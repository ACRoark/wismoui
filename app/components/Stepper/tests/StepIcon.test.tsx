import React from 'react';

import { createSnapshot } from 'testing/utils';

import StepIcon from '../StepIcon';
import { StepperStatus } from '../types';

const expectComponentToMatchSnapshot = (status: StepperStatus): void => {
  const tree = createSnapshot(<StepIcon status={status} />);

  expect(tree).toMatchSnapshot();
};

describe('StepIcon', (): void => {
  it('should display active icon when given active status', (): void => expectComponentToMatchSnapshot(StepperStatus.Active));

  it('should display completed icon when given completed status', (): void => expectComponentToMatchSnapshot(StepperStatus.Completed));

  it('should display error icon when given error status', (): void => expectComponentToMatchSnapshot(StepperStatus.Error));

  it('should display pending icon when given pending status', (): void => expectComponentToMatchSnapshot(StepperStatus.Pending));
});
