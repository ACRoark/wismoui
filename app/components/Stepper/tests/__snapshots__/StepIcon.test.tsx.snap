// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIcon should display active icon when given active status 1`] = `
<i
  className="icon active"
/>
`;

exports[`StepIcon should display completed icon when given completed status 1`] = `
<svg
  className="completed icon"
  fill="currentColor"
  height="1em"
  viewBox="0 0 16 16"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
    fillRule="evenodd"
    stroke="currentColor"
    strokeWidth="1px"
  />
</svg>
`;

exports[`StepIcon should display error icon when given error status 1`] = `
<svg
  className="error icon"
  fill="currentColor"
  height="1em"
  viewBox="0 0 16 16"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M7.002 11a1 1 0 112 0 1 1 0 01-2 0zM7.1 4.995a.905.905 0 111.8 0l-.35 3.507a.552.552 0 01-1.1 0L7.1 4.995z"
  />
</svg>
`;

exports[`StepIcon should display pending icon when given pending status 1`] = `
<i
  className="icon pending"
/>
`;
